common/db/src/lib.rs

```diff
@@ -60,8 +63,58 @@ impl<'a> Database<'a> {
         let db = db::Database::new(database).await?;
         db.execute_unprepared("CREATE EXTENSION IF NOT EXISTS \"pg_stat_statements\";")
             .await?;
+
+        Ok(db)
+    }
+
+    #[instrument(err)]
+    pub async fn bootstrap(database: &config::Database) -> Result<db::Database, anyhow::Error> {
+        let db = Self::setup(database).await?;
+
         Database(&db).migrate().await?;
 
         Ok(db)
     }
+
+    /// Import a database from a provided DB dump.
+    #[instrument(skip(r), err)]
+    pub async fn import<R>(
+        database: &config::Database,
+        mut r: R,
+    ) -> Result<db::Database, anyhow::Error>
+    where
+        R: AsyncRead + Unpin + Send + 'static,
+    {
+        let db = Self::setup(database).await?;
+
+        let mut cmd = PsqlBuilder::new()
+            .dbname(&database.name)
+            .host(&database.host)
+            .port(database.port)
+            .username(&database.username)
+            .pg_password(&database.password.0)
+            .build_tokio();
+
+        let mut child = cmd.stdin(Stdio::piped()).spawn()?;
+
+        let mut stdin = child.stdin.take().ok_or_else(|| anyhow!("Missing stdin"))?;
+        tokio::spawn(async move {
+            io::copy(&mut r, &mut stdin).await.context("copy failed")?;
+
+            let _ = stdin.shutdown().await;
+
+            Ok::<_, anyhow::Error>(())
+        })
+        .await??;
+
+        let _ = child.wait_with_output().await?;
+
+        // some maintenance after import
+
+        db.execute_unprepared(r#"VACUUM FULL ANALYZE"#).await?;
```

Comment:
Should we do 

`REINDEX database trustify`

here as well?

Reply:
Done